app/models/container_template.rb

```diff
@@ -34,11 +34,23 @@ def instantiate(params, project = nil)
                                             :namespace => project
                                           },
                                           :objects    => objects,
-                                          :parameters => params)
+                                          :parameters => params_to_hashes(params))
     create_objects(processed_template['objects'], project)
     @created_objects.each { |obj| obj[:miq_class] = MIQ_ENTITY_MAPPING[obj[:kind]] }
   end
 
+  def params_to_hashes(params)
+    params.to_a.collect do |param|
```

Comment:
removed it, please note that caller anyway needs to pass an array or ActiveRecord collection (just template.container_template_parameters)